src/util/countryDataManager.ts

```diff
@@ -0,0 +1,190 @@
+		await fetch(url)
+			.then((response) => response.json())
+			.catch((err) => logger.debug(err))
+			.then((res) => {
+				return res as CountryImport[];
+			})
+	).sort((a, b) => b.population - a.population);
+
+	logger.debug("initialized the country data");
+}
+
+function CountryDataImportToCountryData(countryImport: CountryImport[]): Country[] {
+	const countrydata: Country[] = [];
+
+	countryImport.forEach((c) => {
+		const newCountry: Country = { ...defaultCountryData };
+
+		Object.entries(extractData).forEach(([key, entry]) => {
+			const value = entry(c);
+			if (value !== undefined) {
+				(newCountry[key as CountryKey] as unknown) = value;
+			}
+		});
+
+		countrydata.push(newCountry);
+	});
+
+	return countrydata;
+	//return countryImport.map(c => Object.fromEntries(Object.entries(extractData).map(([k, v]) => [k, v(c) ?? defaultCountryData[k as CountryKey]])) as unknown as Country);
+}
+
+interface CountryImport {
+	name: { common: string; official: string };
+	cca2: string;
+	tld: string[];
+	unMember: boolean;
+	population: number;
+	capital: string[];
+	languages: object;
+	currencies: object;
+	timezones: string[];
+	region: string;
+	subregion: string;
+	latlng: number[];
+	area: number;
+	maps: { googleMaps: string; openStreetMaps: string };
+	flags: { png: string; svg: string };
+}
+
+// sorting the data
+export function sortCountryDataBy(criteria: CountryKey) {
+	countryData.sort((a, b) =>
+		sortingComparators[typeOfCountryProperty(criteria)](
+			a[criteria] as MainCountryDataTypes,
+			b[criteria] as MainCountryDataTypes
+		)
+	);
+}
+
+const sortingComparators: { [id: string]: (a: MainCountryDataTypes, b: MainCountryDataTypes) => number } = {
+	number: (a, b) => (a as number) - (b as number),
+	string: (a, b) => ((a as string).toLowerCase() <= (b as string).toLowerCase() ? -1 : 1),
+	boolean: (a, b) => ((!a as boolean) && (b as boolean) ? -1 : 0),
+	object: (a, b) =>
+		sortingComparators["string"](
+			(a as string[]).sort(sortingComparators["string"])[0] ?? "",
+			(b as string[]).sort(sortingComparators["string"])[0] ?? ""
+		),
+};
+
+// filtering the data
+export function getFilteredCountryDataBy(
+	criteria: CountryKey,
+	relation: string,
+	value: number | string | boolean
+): Country[] {
+	return countryData.filter((c) =>
+		filteringComparators[typeOfCountryProperty(criteria)](c[criteria] as MainCountryDataTypes, relation, value)
+	);
+}
+
+const filteringComparators: {
+	[id: string]: (
+		value: MainCountryDataTypes,
+		relation: keyof typeof dataRelations,
+		valueCom: number | string | boolean
+	) => boolean;
+} = {
+	number: (a, rel, b) => dataRelations[rel](a as number, b as number),
+	string: (a, rel, b) => dataRelations[rel]((a as string).toLowerCase(), (b as string).toLowerCase()),
+	boolean: (a, rel, b) => dataRelations[rel](a as boolean, b as boolean),
+	object: (a, rel, b) => (a as string[]).map((s) => filteringComparators["string"](s, rel, b)).some((e) => e),
```

Comment:
What is happening in this object comparator, why is it being cast into a string array?
Also this no need to loop twice:
```suggestion
	object: (a, rel, b) => (a as string[]).some((s) => filteringComparators["string"](s, rel, b)),
```